#SOP  #线粒体  #注释
# 1. MITOS网站注释
![[Pasted image 20230308154248.png]]
```shell
打开http://mitos2.bioinf.uni-leipzig.de/index.py
Reference 选择63  
Genetic Code 选择5无脊椎动物or脊椎动物2
点击upload 上传fasta文件
高级选项ncRNA overlap 改100
点击submit 等待结果生成
```
![[Pasted image 20230308155409.png]]
# 2.下载注释结果
![[Pasted image 20230308155436.png]]
```shell
# 下载文件 做trna二级结构
# 要先切换到cluster节点!!!
analysis/annotation/*]$ nohup perl /share/nas6/xul/program/mt/annotation/get_tbl_trnass_from_mitos2.pl -i  "网站链接(点击图片后的详情界面)" && cd trna.structure/ && perl /share/nas6/xul/program/mt/tRNA/draw_tRNA.pl -i trn*/*.svg &

# nohup perl /share/nas6/xul/program/mt/annotation/get_tbl_trnass_from_mitos2.pl -i  "网站链接(点击后)" && mt_tbl2ann_mt.pl  tmp.tbl  >gene.annotation.info && realpath  gene.annotation.info &
# cd trna.structure/ && nohup perl /share/nas6/xul/program/mt/tRNA/draw_tRNA.pl -i trn*/*.svg &
```
# 3. 注释修改
## 3.0 Tips
```shell
# 检查原则（仅供参考，未更新）
1.所有cds的起止密码子
gbk文件里 complement(8446..8590) 互补链
末尾未确定的那种  可以往后延伸1bp  也就是 构建出终止子 优先级大于 是否和trna重叠
2.看重叠,关注负值(尤其是cds的负值,即cds不能与trna重叠)
trna的负值问题不大
cds间可相互重叠
以下4种情况:
若cds与trna方向不同,则不必考虑重叠问题(但是cds+ trna-也还是要考虑)
同向,cds不能与trna重叠,尤其是终点(起点可以落在trna中,但终点不行)
综上,看cds的负值(终点),要改成与trna相接
3.0查找缺失的基因,去除多余的trna
如果缺少trna,考虑只注释cds,这样能去掉cds干扰   每周根据项目总结一下  4228项目
3.删除多余的cds,dloop,确保基数为cds13 trna22 rrna2,共37行
http://rna.tbi.univie.ac.at/cgi-bin/RNAWebSuite/RNAfold.cgi
RNAfold  < test.fa  > tmp.fold
RNAplot -o svg   < tmp.fold && display *.svg
图片改名
添加trna-反密码子(对该氨基酸所对应密码子的反向互补)
网页版参数勾选 1 3 3
4.修改基因名字,把前面的计数  后缀为0 1都去掉
5.根据参考找r1及r2,一般两端直接与两侧基因相接
6.修改dloop区,共38行
7.把生成的trna图片转换后 一并放入文件夹
svg2xxx -t pdf trnH.svg

# xul的一些程序
$ cp_annotation_one_gene_by_ref_gbk2.pl -i1 Thainanus_FULLMT.fsa  -i2 ref_ann/gbk/NC_050664.1.gbk -g COX1 # 不看蛋白用叶绿体的找基因或者取序列
$ cp_add_gene_sequence.pl  -i ref_ann/fasta/NC_050664.1.fasta -p 74-1039:+  # 只取序列不看蛋白用叶绿体的就行
$ mt_add_gene_sequence.pl  -i ref_che/fasta/NC_038207.1.fasta -p 13784-13834 -g ?  # 要看蛋白的话,用线粒体,需要指定密码子表
$ blastn -query tmp.fa  -subject *.fsa -outfmt 6 # blastn	DNA
```
## 3.1 具体修改流程(详见 [[05线粒体流程2-2注释答疑]])
```shell
0.下载参考
$ vim list_ann
$ down- ann && cd ref_ann/gbk && 2ann- && cd -
# for i in *.gbk ;do echo $i; cp_gbk2ann.pl -i $i -o $i.ann ;done # 抽取参考注释

1.解析注释结果（真菌的话，不执行这一步，直接执行步骤2-1）
$ vim gene.info # 复制网页表格（包括表头）至gene.info文件
$ python3 /share/nas1/yuj/script/mitochondrion/annotation/mt_parse_info_table.py -i gene.info -o gene.annotation.info2 -n 2/5 # 2/5选择密码子表

2-1.查看基因排列,根据参考修改注释(真菌)
$ grep -v 'lagli' gene.annotation.info | grep -v 'dpo' | grep -v 'giy' > gene.annotation.info2
$ grep 'CDS' gene.annotation.info2 
$ for i in `grep 'CDS' gene.annotation.info2 | awk '{print $2}'` ;do fu_add.py  -i *.fsa -p $i -f;done   > sample_ann.log

2.查看基因排列,根据参考修改注释
$ grep 'CDS' gene.annotation.info2
$ grep 'CDS' ref_ann/gbk/登录号.gbk.ann

$ for i in `grep 'CDS' gene.annotation.info2 | awk '{print $2}'` ;do mt_add.py -n 2/5 -i *.fsa -p $i;done   > sample_ann.log
$ for i in `grep 'CDS' ref_ann/gbk/登录号.gbk.ann | awk '{print $2}'` ;do mt_add.py -n 2/5 -i ref_ann/fasta/登录号.fasta -p $i;done   > ref_ann.log

# 以下查找cox1基因
$ vim cox1
$ tblastn -query cox1 -subject *.fsa -outfmt 6
$ mt_add.py -n 5 -i *.fsa -p # 检查基因，分段的话 -p参数的位置要加双引号

3.没有问题的话,生成gene.annotation.info
$ cp gene.annotation.info2 gene.annotation.info
# 有问题，修改序列起点
$ python3 /share/nas1/yuj/script/mitochondrion/annotation/mt_move_gene_pos.py -i gene.annotation.info2 -o gene.annotation.info -n2 49 -m 17121 # 平移排序注释 帮助信息见-h

PS：trna二级结构有问题,新建一个文件夹进行生成,不要在之前的文件夹里操作,因为会覆盖注释文件！！！
# 如果trna有问题         用以下脚本查找二级结构
$ python3 /share/nas1/yuj/script/mitochondrion/annotation/mt_trnascan_ss_2_rnaflod.py -i *.fsa -o trna.structure/trn # 脚本会输出下一步要做的操作
$ RNAfold  < test.fa  > tmp.fold
$ RNAplot -o svg   < tmp.fold && display *.svg
```
## 3.2 真菌查找单个基因(以查找nd5为例)
```shell
以查找nd5为例:
$ for i in /share/nas1/yuj/project/GP-20220920-4939_20221107/1/ref_tre/out/cds/*;do grep -A 1 -i 'nd5'  $i >> nd5_all.fa;done

$ wc -l nd5_all.fa 
104 nd5_all.fa

$ for i in {1..104};do if [ $(($i%2)) -eq 0 ];then sed -n ""$i"p" nd5_all.fa > $i"_nd5";blastn -query $i"_nd5" -subject *.fsa -outfmt 6;fi;done

$ blastn -query cox1_all.fa -subject *.fsa -outfmt 6 > blastn_cox1

从结果里挑选最符合的
```
## 3.3 下一步流程
```shell
1.配置文件
编号目录]cp /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/cp.anno.config.yaml ./ann.cfg && readlink -f ./ann.cfg
# 修改配置文件,写参考

2.跑流程
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/mt_annotation_pip.pl -i ann.cfg -g 2/5 -c
# 第一遍流程加-c参数,好用来检查,检查gene.bed.xls
# 如果确信1.2步骤里做的没问题，可以不加-c，一遍过

/share/nas6/xul/program/mt/bin/mt_get_gene_bed_more.pl -g $gencode -i $indir/annotation/$sample/$sample.gbk > $indir/annotation/$sample/gene.bed.xls
# 程序结果
```
# 4.整理结果
```shell
$ perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/mt_pip_dir.pl -i analysis
```
# 5.整理报告
```shell
$ cp /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/report.cfg ./report.cfg && realpath  report.cfg # 修改
 
# 图片路径 /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/src/tmp

# 标准分析,不加-n也行
$ perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/report2xml.xianliti.pl  -id complete_dir/ -cfg report.cfg -n

# 标准分析+高级分析 / 标准分析用也行
$ perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/report2xml.xianliti.pl  -id complete_dir/ -cfg report.cfg	

# 标准分析，不包含测序
$ perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/report2xml.xianliti2.pl -id complete_dir/ -cfg report.cfg	

# 三代辅助组装
$ perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/report2xml.xianliti.3dai.pl   -id complete_dir/ -cfg report.cfg	
```
# 6.补充
## 6.1质控
```shell
# 组装 质控
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/asmqc/mtDNA_asmqc.pl -i 组装结果  -1 clean*1 -2 clean*2 -p sample -q 物种.gbk  -o asmqc -g 2 -r 参考.gbk

# 测序数据 质控
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/src/fastq_qc.pl -i analysis/samples.reads.txt -o complete_dir/01Rawdata 

3490质控出问题,因为忘记改记录原始数据路径的文件了
```
## 6.2圈图Chloroplot
```bash
1.windows分步骤
$ Rscript E:\OneDrive\jshy信息部\Script\chloroplast\R\Win_Chloroplot.R F:\  F:\4939_202301\Ustilago_esculenta_MT10.gbk Ustilago_esculenta_MT10 mt
$ 复制pdf到服务器
$ convert -density 600 *.circular.pdf  Ustilago_esculenta_MT10.dpi300.circular.png

2.
使用Chloroplot（https://irscope.shinyapps.io/Chloroplot/）制作线粒体基因组图谱，如下图所示：
注：正向编码的基因位于圈内侧，反向编码的基因位于圈外侧。内部的灰色圈代表GC含量。

3.流程里对这部分的文字描述
#-----4.3.2 线粒体基因组图谱
&EMPTY_TAG('h3','4.3.2 线粒体基因组图谱','','type1');
&EMPTY_TAG('p','使用Chloroplot（https://irscope.shinyapps.io/Chloroplot/）制作线粒体基因组图谱，如下图所示：','type1');
&piclist("图$pid 线粒体基因组图谱",'注：正向编码的基因位于圈内侧，反向编码的基因位于圈外侧。内部的灰色圈代表GC含量。',"$indir/04Annotation/*/ogdraw/*dpi300.circular.png");
```
## 6.3其他问题
```shell
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/src/rscu.pl -i1 Mm_S3/features/Mm_S3.cds -i2 Mm_S3/features/Mm_S3.pep -g 5 -o Mm_S3/rscu_analysis/rscu.stat.xls
```
# 7.批量操作 
```shell
# for i in $( ls /share/nas1/yuj/project/20211229/GP-20211206-3811/analysis/assembly) ;do echo $i

# 显示site文件的第一列与第二列,以空格分开
for i in `awk -F "\t" '{print $1 $2}' site.txt`;do echo $i;done

# 批量写入shell命令到文件,site.txt里是网址,执行完再单独写到每个.sh文件中
for i in `awk '{print $1}' site.txt`;do echo $i && echo -e "nohup perl /share/nas6/xul/program/mt/annotation/get_tbl_trnass_from_mitos2.pl -i  \"$i\" && mt_tbl2ann_mt.pl  tmp.tbl  >gene.annotation.info && realpath  gene.annotation.info &" >> shell.txt;done 

# 测试
for i in {g3,j1,j2,j3,l1,l2,l3,s1,s2,s3};do echo $i && cat $i.sh;done 

# 批量执行已写好的脚本
for i in {Mm_G3,Mm_J1,Mm_J2,Mm_J3,Mm_L1,Mm_L2,Mm_L3,Mm_S1,Mm_S2,Mm_S3};do echo $i;cd /share/nas1/yuj/project/202201/GP-20211206-3816/analysis/annotation/$i &&pwd && sh /share/nas1/yuj/project/202201/GP-20211206-3816/$i.sh;done

# 批量整理trna二级结构
for i in {Mm_G3,Mm_J1,Mm_J2,Mm_J3,Mm_L1,Mm_L2,Mm_L3,Mm_S1,Mm_S2,Mm_S3};do echo $i;cd /share/nas1/yuj/project/202201/GP-20211206-3816/analysis/annotation/$i/trna.structure/ && pwd &&  perl /share/nas6/xul/program/mt/tRNA/draw_tRNA.pl -i trn*/*.svg && pwd;done

# 批量blast比对
for i in {Mm_J2,Mm_J3,Mm_L1,Mm_L2,Mm_L3,Mm_S1,Mm_S2,Mm_S3};do echo $i;blastn -query /share/nas1/yuj/project/202201/GP-20211206-3816/r1.fa  -subject /share/nas1/yuj/project/202201/GP-20211206-3816/analysis/assembly/$i/finish/*.fsa -outfmt 6;done

# 看组装完的contig深度   和红原鸡线粒体基因组比对   根据深度调整
blastn -query allref.fa  -subject *.fsa -outfmt 6 

# 批量显示测序数据量
for i in $( ls /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/analysis/assembly/) ;do echo $i; ll /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/analysis/assembly/$i/pseudo/$i/1_Trimmed_Reads/$i.trimmed_P1.fq && ll /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/analysis/assembly/$i/pseudo/$i/1_Trimmed_Reads/$i.trimmed_P2.fq ;done

# 批量显示比对结果
for i in $( ls /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/analysis/assembly/) ;do echo $i; blastn -query /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/ref_ass/all_ref.fa  -subject /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/analysis/assembly/$i/pseudo/$i/1_Trimmed_Reads/uni/assembly.fasta -outfmt 6 ;done

# 批量质控
for i in $( ls analysis/assembly/) ;do echo $i; perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/asmqc/mtDNA_asmqc.pl -i ref_ass/fasta/HQ857211.1.fasta  -1 /share/nas1/seqloader/xianliti/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/2.clean_data/$i/$i_*_R1_001.fastq.gz -2 /share/nas1/seqloader/xianliti/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/2.clean_data/$i/$i_*_R2_001.fastq.gz -p Gallus_gallus -q /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/ref_ass/gbk/HQ857211.1.gbk  -o /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/analysis/annotation/$i/asmqc -g 2 -r /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/ref_ass/gbk/MT800385.1.gbk;done
```