>结果里会少最后一个物种,先跑一遍流程,再单独跑程序生成cgview kaks
>cgview注意顺序,如果可以的话
>mauve记得改顺序
>进化树见进化树部分
# 1.参考基因组数据准备
## 1.1下载
```shell
# 复制表格中的物种 至 tmp_adv文件(以+号为分割) 
编号目录]$ awk -F "+" '{ print $2 }' tmp_adv | awk '$1=$1' > list_adv && cat list_adv && down- adv && cd ref_adv/ && cat fasta/* > allref.fa && cd ../
```
## 1.2改名
```shell
# 利用awk提取一下名字 cd ref_adv/fasta && for i in *.fasta;do awk '{if(NR==1) print $0}' $i;done | awk -F ">" '{ print $2 }' | awk -F ".1_" '{ print $2"""_"""$1""".1" }' && cd -
# print a""b >>> 结果 ab

# 给gbk改名
$ cd ref_adv/fasta && for i in *.fasta;do rename               `awk '{if(NR==1) print $0}' $i | awk -F ">" '{ print $2 }' | awk -F ".1_" '{ print ""$1""".1" }'`       `awk '{if(NR==1) print $0}' $i | awk -F ">" '{ print $2 }' | awk -F ".1_" '{ print $2"""_"""$1""".1" }'`   ../gbk/""`awk '{if(NR==1) print $0}' $i | awk -F ">" '{ print $2 }' | awk -F ".1_" '{ print ""$1""".1" }'`.gbk  ;done  && cd ../../

# 给fasta改名
$ cd ref_adv/fasta && for i in *.fasta;do rename              `awk '{if(NR==1) print $0}' $i | awk -F ">" '{ print $2 }' | awk -F ".1_" '{ print ""$1""".1" }'`       `awk '{if(NR==1) print $0}' $i | awk -F ">" '{ print $2 }' | awk -F ".1_" '{ print $2"""_"""$1""".1" }'`   ../fasta/""`awk '{if(NR==1) print $0}' $i | awk -F ">" '{ print $2 }' | awk -F ".1_" '{ print ""$1""".1" }'`.fasta  ;done  && cd ../../
```
# 2.0真菌分析
```shell
ref_adv]$ python3 /share/nas1/yuj/script/mitochondrion/fungi/fungi_from_gbk_get_cds_V1.0.py -i gbk/  -o out  -d # -d表示不去重复

```
# 2.高级分析
```shell
1.获取高级分析配置
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/advance/advanced.yaml.final.pl -i analysis/ -f ref_adv/fasta/ -g ref_adv/gbk/

-tree ref_tree/all.fa
# 建议把高级分析物种按物种名排序,也就是改文件名

2.主流程
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/advance/advanced_pip.v1.pl -i advanced.config.yaml -g 2/5 
# 跑流程    2是脊椎动物  5是无脊椎动物
```
# 3.检查高级分析结果
## 3.1cgview圈图
```shell
#对应该步命令,但该程序会生成final.xml文件,覆盖掉修改的参数,如高宽参数
cd analysis]
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/current/cgview_comparison/cgview_cmp_gbk.pl -i ../ref_adv/sample/gbk/Mm_G1.gbk -c ../ref_adv/gbk/*.gbk -o cgview &

1.若有重叠
cgview]$ vi maps/cgview_xml/mtDNAsmall.final.xml
		height="1050" width="1050"  修改高宽参数

cd cgview]$ java -jar -Xmx1500m /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v1.0/cgview_comparison/etc/cgview_comparison_tool/bin/cgview.jar -i maps/cgview_xml/mtDNAsmall.xml -o mtDNA.cgview_cmp.svg -f svg

&& svg2xxx -t png -dpi 300 mtDNA.cgview_cmp.svg & svg2xxx -t pdf mtDNA.cgview_cmp.svg & 
```
## 3.2kaks选择压力分析
```SHELL
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/advance/kaks/kaks.each.two.pl

cd kaks && ll */*xls
```
## 3.3Mauve基因组比对
```shell
Mauve & # 用第二种比对方式
```
## 3.4phytree
```shell
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/advance/phytree/raxml_phytree.pl
```
## 3.5全局比对(snp indel)    一般不做
```shell
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/advance/global_align/global_align.pl
# 同叶绿体

perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/script/global_align/global_align.pl -i 组装出的基因组(一般是第一个) -a 一般是第一个物种注释 -r 参考基因组/其他物种.fa -o global_align
```
## 3.6核酸多样性分析(线粒体缺此分析程序)
```shell
输入所有gbk所在的目录

perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/advance/pi.analysis/pi.no.ir.analysis.pl -i /share/nas1/yuj/project/GP-20211206-3816/ref_adv/gbk -p Mm_G1 -o /share/nas1/yuj/project/GP-20211206-3816/analysis/pi

perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/script/pi.analysis/pi.no.ir.analysis.pl -i /share/nas1/yuj/project/GP-20211206-3816/ref_adv/allinonegbk -p Mm_G1 -o /share/nas1/yuj/project/GP-20211206-3816/analysis/pi
```
# 4.整理结果
```shell
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/mt_pip_dir.pl -i analysis

0.原始数据统计 可能出问题
/share/nas6/zhouxy/modules/ngsqc/ngsqc_bgi -1 fq1 -2 fq2 -k 物种名 -o ./(华大可用.)
/share/nas6/zhouxy/biosoft/bin/ngsqc -1 fq1 -2 fq2 -k 物种名 -o ./ (华大不能用)
Rscript /share/nas6/xul/program/reseq/v1.1/script/ngsqc.R --base *.atgc --qual *.qual  --out ./

			sub qc{
				system "/share/nas6/zhouxy/biosoft/bin/ngsqc -1 $_[1] -2 $_[2] -k $_[0] -o . ";
				system "Rscript /share/nas6/xul/program/reseq/v1.1/script/ngsqc.R --base $_[0].atgc --qual $_[0].qual  --out $_[0]";
			}

			for my $svg (@svg){
				system "svg2xxx -t pdf $svg.acgtn.svg";
				system "svg2xxx -t png -dpi 600 $svg.acgtn.svg";
				system "svg2xxx -t pdf $svg.quality.svg";
				system "svg2xxx -t png -dpi 600 $svg.quality.svg";
```
# 5.生成报告
```shell
# 修改
cp /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/report.cfg ./report.cfg && realpath  report.cfg 

# 标准分析,不加-n也行
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/report2xml.xianliti.pl  -id complete_dir/ -cfg report.cfg -n  

# 标准分析+高级分析 / 标准分析用也行
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/report2xml.xianliti.pl  -id complete_dir/ -cfg report.cfg

# 标准分析，不包含测序
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/report2xml.xianliti2.pl -id complete_dir/ -cfg report.cfg	

# 三代辅助组装
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/report2xml.xianliti.3dai.pl   -id complete_dir/ -cfg report.cfg 

0.容易出问题的质控
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/asmqc/mtDNA_asmqc.pl -i 组装结果  -1 clean*1 -2 clean*2 -p 物种名 -q 物种.gbk  -o asmqc -g 2 -r 参考.gbk
# 3490质控出问题,因为忘记改记录原始数据路径的文件了
```
