---
title: 05线粒体分析流程-完整版
description: 
published: true
date: 2023-03-09T03:59:01.563Z
tags: 
editor: markdown
dateCreated: 2023-03-09T03:58:57.545Z
---

# Assembly
注意定期更换参考基因组库 
/share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/assembly/etc/Fast-Plast/MitoF.pl  /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/assembly/dat/
```SHELL
$ perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/assembly/src/get_ass_cfg.pl -i data  # 记得改物种名!

$ nohup perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/assembly/mt.assembly.pip.pl -i ass.cfg & 

/share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/assembly/etc/Fast-Plast/MitoF.pl  # 修改线粒体数据库

调整开头为trna phe
$ perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v1.0/assembly/src/start.adjust.pl

批量延伸：
$ for i in * ;do echo "cp  $i/pseudo/$i/4_Afin_Assembly/filtered_spades_contigs.fsa   $i/pseudo/$i/4_Afin_Assembly/test.fa && extend.auto.pl -i1 $i/pseudo/$i/4_Afin_Assembly/test.fa -i2 $i/pseudo/$i/1_Trimmed_Reads/*P1.fq -sr ";done  > extend.sh
$ thread.pl extend.sh 6
	
批量比对组装：
$ for i in * ;do echo "cp_bowtie.align.pl  -1 $i/pseudo/$i/1_Trimmed_Reads/*P1.fq -2 $i/pseudo/$i/1_Trimmed_Reads/*P2.fq -i $i/pseudo/$i/4_Afin_Assembly/test.fa -o $i/pseudo/$i/4_Afin_Assembly/bow/ && unicycler.py -1 $i/pseudo/$i/4_Afin_Assembly/bow/map_pair_hits.1.fq -2 $i/pseudo/$i/4_Afin_Assembly/bow/map_pair_hits.2.fq -o $i/pseudo/$i/4_Afin_Assembly/bow/uni/" ;done  > bow_and_ass.sh
$ thread.pl bow_and_ass.sh 6
```

# Annotation
```SHELL
--download tbl and trna ss
nohup perl /share/nas6/xul/program/mt/annotation/get_tbl_trnass_from_mitos2.pl -i  "`cat web`" && mt_tbl2ann_mt.pl  tmp.tbl  >gene.annotation.info && realpath  gene.annotation.info & 

--提前做trna二级结构：
cd trna.structure/ && nohup perl /share/nas6/xul/program/mt/tRNA/draw_tRNA.pl -i trn*/*.svg &
display final_tRNA/trnS1.svg 

--填写配置文件
cp /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/cp.anno.config.yaml ./ann.cfg && readlink -f ./ann.cfg
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/mt_annotation_pip.pl	-i ann.cfg
# 记得改dloop区的位置,需要把每个tbl放到对应的annotation下并转成 gene.annotation.info

--tRNA structure	# ViennaRNA
RNAfold  < test.1.fa > test.fold
RNAplot --pre   "1 69 8 0 255 0  omark"<  test.fold  && convert -background white -fill black -pointsize 30 -font helvetica -draw 'text 20,35 "trnF"' [gene=trnF-GAA]_ss.ps -flatten  test.png
RNAfold  < test.fa  > tmp.fold &&  RNAplot -o svg   < tmp.fold && display 1_ss.svg

perl /share/nas6/xul/program/mt/tRNA/draw_tRNA.pl -i trn*/*.svg -c && display final_tRNA/trnS1.svg
vim final_tRNA/trnS1.svg
perl /share/nas6/xul/program/mt/tRNA/draw_tRNA.pl -i trn*/*.svg


--tbl2ann
perl /share/nas6/xul/program/mt/bin/mt_tbl2ann_mt.pl -i tmp.tbl >gene.annotation.info && readlink -f gene.annotation.info

--ann2gbk
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/src/to_genbank.pl
	/note="TAA stop codon is completed by the addition of 3' A residues to the mRNA"
	/transl_except=(pos:complement(6398),aa:TERM)
	/note="internal stop codon"
	<1..20 or 3..>44  /note="stop codon not determined"  /note="start codon not determined" 

--老的注释,可以用来比较
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v1.0/annotation/bin/mitochondrion_annotaion.pl

--CDS
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v1.0/annotation/src/blast.anno.mrn.pl

--找近缘的序列
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v1.0/annotation/src/blast.dat.pl

--质控（asmqc）
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/asmqc/mtDNA_asmqc.pl
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/asmqc/src/cmp_genebank.pl -r cmp.gbk -q ../HH.gbk  -o cmp.genecov_stat.xls

--dotplot
perl  /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/asmqc/src/dotplot.pl -r refseq.fa -q ../../../assembly/sorex/finish/sorex_FULLMT.fsa -p cmp -o .

--cmpgbk
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/asmqc/src/annotation_mtDNA_genbank.pl

--cmp.genecov_stat.xls
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/asmqc/src/cmp_genebank.pl -r ../../../../ref/gbk/NC_037859.1.gbk -q ../sorex.gbk  -o cmp.genecov_stat.xls

--coverage_plot
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/asmqc/src/mt2circos.pl -g cmp.gbk -b mapping/sample.sort.bam -o coverage_plot

--cov stat 
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/asmqc/src/coverage_stat.pl -i . -o . -p

--rawdata
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/src/fastq_qc.pl -i sample.reads.txt -o rawdata

--重复片段

--ogdraw
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/etc/plot_ogdraw_cpgenome_genbank/ogdraw.pl

--feature
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/src/extract.feature.pl

--GC
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/src/base.skew.pl

--all_GC
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/src/all.sample.gc.pl  -i $outdir -o $outdir/sample.gc.stat.xls

--gbk2tbl
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/src/gb2tbl.pl

--rscu
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/src/plot.rscu.bar.pl
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/src/plot_rscu_table.pl
```

# Advance
```shell
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/advance/advanced.yaml.final.pl -i analysis/ -f ref_ad/fasta/ -g ref_ad/gbk/ -tree ref_tree/all.fa
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/advance/advanced_pip.v1.pl -i advanced.config.yaml -g 

--cgview
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/current/cgview_comparison/cgview_cmp_gbk.pl -i *.gbk -c ref/gbk/*.gbk -o cgview &
java -jar -Xmx1500m  /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v1.0/cgview_comparison/etc/cgview_comparison_tool/bin/cgview.jar -i maps/cgview_xml/mtDNAsmall.xml -o mtDNA.cgview_cmp.svg -f svg && svg2xxx -t png -dpi 300 mtDNA.cgview_cmp.svg & svg2xxx -t pdf mtDNA.cgview_cmp.svg & 

--phytree
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/advance/phytree/raxml_phytree.pl

--kaks
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/advance/kaks/kaks.each.two.pl

--Mauve

```

# 整理结果
```sehll
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/mt_pip_dir.pl -i analysis
```

# 报告
```shell
cp /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/report.cfg ./report.cfg && realpath  report.cfg

perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/report2xml.xianliti.pl  -id complete_dir/ -cfg report.cfg -n  # 标准分析
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/report2xml.xianliti.pl  -id complete_dir/ -cfg report.cfg	# 标准分析+高级分析

perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/report2xml.xianliti2.pl -id complete_dir/ -cfg report.cfg	# 标准分析，不包含测序
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/html_report/report2xml.xianliti.3dai.pl -id complete_dir/ -cfg report.cfg	# 三代辅助组装
```



```shell
for i in *.gbk ;do perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/src/extract.feature2.pl  -i   $i -o seq  -c ;done 
for i in *.gbk ;do perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/src/extract.feature2.pl  -i   $i -o seq  -r ;done


prefix="../../gene_tree" && \
mkdir $prefix 

cd seq && \
cat `ls * |grep -i d1 `     > ../$prefix/nd1.fa && \
cat `ls * |grep -i d2 `     > ../$prefix/nd2.fa && \
cat `ls * |grep -i d3 `     > ../$prefix/nd3.fa && \
cat `ls * |grep -i d4 `   > ../$prefix/nd4.fa && \
cat `ls * |grep -i d4l `    > ../$prefix/nd4l.fa && \
cat `ls * |grep -i d5 `     > ../$prefix/nd5.fa && \
cat `ls * |grep -i d6 `     > ../$prefix/nd6.fa && \
cat `ls * |grep -i x1 `     > ../$prefix/cox1.fa && \
cat `ls * |grep -i x2 `     > ../$prefix/cox2.fa && \
cat `ls * |grep -i x3 `     > ../$prefix/cox3.fa && \
cat `ls * |grep -i cob `    > ../$prefix/cob.fa && \
cat `ls * |grep -i cytb `  >> ../$prefix/cob.fa && \
cat `ls * |grep -i atp6 `   > ../$prefix/atp6.fa && \
cat `ls * |grep -i atp8 `   > ../$prefix/atp8.fa && \
cat `ls * |grep -i rrnL `   > ../$prefix/rrnL.fa && \
cat `ls * |grep -i rrnS `   > ../$prefix/rrnS.fa && \
cd ../$prefix && \
mkdir aln && \
for i in *fa;do mafft --auto $i > aln/$i.aln;done
cd aln && \
mkdir trim && \
for i in *.aln ;do /share/nas6/xul/soft/trimal/trimal-1.4.1/source/trimal -in  $i -out  trim/$i.trim.fasta ;done

perl /share/nas6/xul/program/chloroplast/phytree/connection_head_tail.pl -i *.aln

nohup perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/advance/phytree/raxml_phytree.pl -i Single_gene_seq.aln -aln -o .. -p 13_cds_2_rrn


# CDS,trna,rrna
提取序列，需要基因组结构相同
/share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/script/pi.analysis/src/extract.dir.cds_tmp_for_mt.pl
```
