# 一.准备数据
```shell
1. 测序所在路径，一般有多个文件夹，每个样品一个文件夹

2. 基因组fasta文件，有时需要下载，有时客户提供，还有常见的或者我们做过的可以到 /share/nas6/pub/genome中找找

3. 基因组注释文件，gff3格式。

4. 染色体列表，txt文件，一列。存放序列名称，因为有的fasta文件中有很多序列，该文件用于定义某些图片中需要展示的序列
一般是保留染色体级别的序列，如果该基因组都是scaffold，那么就保留几个比较长的序列名称，十个左右就行。
序列名称为fasta文件中">"后面的连续非空字符。
grep ">" Gardenia_genome/Gardenia_v2.fasta > chrlist.txt
fl Gardenia_genome/Gardenia_v2.fasta | sort -k 1 -n # -n:从小到大  不加-n：从大到小
# 发现有11个染色体级别的
fl Gardenia_genome/Gardenia_v2.fasta | sort -k 1 | head -n 11 | awk '{print $2}'

5. 功能注释结果的整合文件

6. 分组
```

# 二.功能注释（可放后台  如果已经做过了，可以跳过该步骤，nas6中的基因组一般都做过这个分析）
```SHELL
1. 获取转录本序列文件，fasta格式的核酸序列。
gffread -x cds.fa -g genome.fa  in.gff3
遇到错误就删删删
例子:
grep -n "rna-A564_p001"  GCF_004118075.2_ASM411807v2_genomic.gff | wc -l
for i in {1..6};do echo $i;sed -i '649514d' GCF_004118075.2_ASM411807v2_genomic.gff;done
# cds要用id来命名，重测序这块分析可能不用管

2.获取最长转录本
有的基因可能存在多个转录本，所以需要过滤下，根据上述提取出来的基因名称判断哪些是同一个基因的，一般id后面有.1 .2这种，就属于一个基因的不同转录本。
具体还要看提取出来的格式。
参考程序：
perl  /share/nas6/zhouxy/functional_modules/cnvfmt_trans2longest/fasta2longest.pl  -i cds.fa  -o filter.cds.fa(旧的程序)
perl /share/nas6/zhouxy/functional_modules/cnvfmt_trans2longest/fasta2longest_with_gff3.pl -i -g -o (新程序)
# perl /share/nas6/zhouxy/functional_modules/cnvfmt_trans2longest/fasta_trans_to_geneid_format_convert_ncbi.pl 格式转换  没用过

3. 填写功能注释的配置文件，以植物为例：
cp /share/nas6/pub/pipline/gene-function-annotation/v1.3/profile/plant_QAll.cfg .
修改第一行，改成filter.cds.fa的 ！绝对路径！。下面的nr库需要根据情况改成单子叶的或者双子叶的。

4. 运行主程序
nohup perl /share/nas6/pub/pipline/gene-function-annotation/v1.3/Gene_Func_Anno_Pipline.pl --nr --swissprot --cog --kog --kegg --pfam --GO --cfg *_QAll.cfg --od Basic_function &

程序运行完后需要用到的文件就是：Basic_function/Result/All_Database_annotation.xls 这个文件，用于填写下面的配置文件
```

# 三.主流程
>/share/nas6/zhouxy/pipline/pop-pip/current/pop_pip_v5.pl
```shell
配置文件：
perl /share/nas6/xul/program/reseq/create_profile_for_reseq_pop.pl   
-i      <in dir>        contain all.gz file or fq or fastq        # 输入包含测序文件所在的路径（文件夹，不是具体文件路径），如 path/clean/*/
-s      ref genome    default #/share/nas6                # 输入参考基因组fasta文件
-g      ref chr list       default #/share/nas6            # 输入染色体列表
-c      ref anno       Gene_Anno/Result/All_Database_annotation.xls  default #/share/nas6    # 输入功能注释的结果Basic_function/Result/All_Database_annotation.xls，如果功能注释没做完，这个可以先不填。
-gff    ref gff       default #/share/nas6                # 输入gff文件
-Is     insert_size,default 100-500                    # 插入片段大小，一般不用改
-p      Prefix  default XM                        # 物种名简拼，不改也行
# 手动添加分组 详见35个栀子分析 后续写程序改进一下

主流程：
$ nohup perl /share/nas6/zhouxy/pipline/pop-pip/current/pop_pip_v5.pl -i ref_reseq_config.yaml -o analysis &

gwas：/share/nas6/zhouxy/pipline/gwas_vcf_pip/current/gwas_ped_pip_v2.pl

整理结果：
# perl /share/nas6/zhouxy/pipline/pop-pip/v1.1/script/resultdir/resultDir.pl -i analysis/ -o complete 原版有点问题
$ perl /share/nas1/yuj/pipline/pop-pip/v1.1/script/resultdir/resultDir.pl -i analysis/ -o complete

报告配置：
$ cp /share/nas6/xul/project/ddrad/GP-20210729-3250_ningxiadaxue_211samples_niu_ddrad/pop_html.cfg . && realpath pop_html.cfg

生成报告:
# perl /share/nas6/zhouxy/pipline/pop-pip/v1.1/html_report/pop_report2html.pl -id complete/ -cfg pop_html.cfg 原版有点问题
$ perl /share/nas1/yuj/pipline/pop-pip/v1.1/html_report/pop_report2html.pl -id complete/ -cfg pop_html.cfg
```

# 四.有参 指纹图
```bash
指纹图等上面做完后单独做

# 以/share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/vcf_filter/samples.pop.snp.recode.vcf为例
$ cd analysis/vcf_filter
# analysis/variation_dir/variants_anno_dir 有点大
# analysis/vcf_filter/samples.gatk.snp.m2M2.mis0.2.mac3.4dtv.vcf 不确定

配置文件：
$ cp /share/nas6/zhouxy/project/ddrad/GP-20220524-4388-1_guizhoudaxue_chentaolin_48samples_ddrad/kasp-develop.config.yaml kasp-develop.config.yaml
修改配置文件
   
流程：
$ perl /share/nas6/zhouxy/pipline/kasp/kasp-develop/v1.1/kasp-develop.pl -i kasp-develop.config.yaml -o kasp_analysis

整理结果：
$ perl /share/nas6/zhouxy/pipline/kasp/kasp-develop/v1.1/bin/resultsdir/resultsdir_v2.pl -i kasp_analysis/ -o kasp_complete_dir

报告：
$ cp /share/nas6/zhouxy/pipline/kasp/kasp-develop/v1.1/html_report/kasp_report.cfg . && realpath  kasp_report.cfg
$ perl /share/nas6/zhouxy/pipline/kasp/kasp-develop/v1.1/html_report/kasp_Web_Report.pl -id kasp_complete_dir/ -cfg kasp_report.cfg
```

# 五.FAQ
## 5.1 gatk_VQSR.failed.cmds
>[!info] 
> 运行`/share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/.cmds_dir/gatk_VQSR.failed.cmds`出现`Exception in thread "main" java.lang.OutOfMemoryError: GC overhead limit exc`
> > [!done] 调大jvm的内存参数
> > ```bash
> > --java-options "-Xmx64g -Xms64g" 
> > ```
## 5.2 plot-vcfstats可视化
> [!info]
> ```txt
> Parsing bcftools stats output: /share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/variation_dir/variants_stat/samples.var.stat
> Plotting graphs: python3 plot.py
> Creating PDF: pdflatex summary.tex >plot-vcfstats.log 2>&1
> The command exited with non-zero status, please consult the output of pdflatex: /share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/variation_dir/variants_stat/samples.varplot-vcfstats.log
>  at /share/nas6/zhouxy/biosoft/bcftools/current/bin/plot-vcfstats line 111.
>  main::error("The command exited with non-zero status, please consult the o"...) called at /share/nas6/zhouxy/biosoft/bcftools/current/bin/plot-vcfstats line 2083
>  main::create_pdf(HASH(0xbb2d98)) called at /share/nas6/zhouxy/biosoft/bcftools/current/bin/plot-vcfstats line 72
> ```
> log
> ```txt
> /share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/variation_dir/variants_stat/samples.var/plot-vcfstats.log
> 
> ! LaTeX Error: File 'multirow.sty' not found.
> ```
> > [!done] 换用tectonic summary.tex
> > ```bash
> > cd /share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/variation_dir/variants_stat
> > cd samples.var/
> > python3 plot.py # pdflatex --version
> > conda create -n bcftools -y
> > conda activate bcftools
> > conda install -c conda-forge Tectonic # tectonic --version
> > tectonic summary.tex # 生成summary.pdf
> > ```

> [!tip]
> plot-vcfstats脚本在bcftools安装目录misc文件夹下，这是一个perl脚本，会调用python3绘图模块Matplotlib。如果没有安装该模块，可以通过pip3命令`pip3 install -U matplotlib`进行安装。如果没有pip3，通过`sudo apt-get install python3-pip`命令安装。
> plot-vcfstats生成pdf文件还需要pdf-latex，如果系统没有安装latex，通过`sudo apt-get install texlive-full`进行安装。

> [!quote]
> 1."python - Dependencies plot-vcfstats in conda environment - Stack Overflow." https://stackoverflow.com/questions/69659849/dependencies-plot-vcfstats-in-conda-environment, 访问时间 1 Jan. 1970.
> 
> 2."Plot-VCF issues · Issue #1697 · samtools/bcftools · GitHub." https://github.com/samtools/bcftools/issues/1697, 访问时间 1 Jan. 1970.
> 
> 3."Impossible to include a PDF image with '.' in filename · Issue #187 · tectonic-typesetting/tectonic · GitHub." https://github.com/tectonic-typesetting/tectonic/issues/187, 访问时间 1 Jan. 1970.

## 5.3 vcf2plink.cmds -- Too few valid variants for --indep-pairwise
```bash
# vcf2plink.cmds 这一步出错,其中第一条命令产生最终的samples.plink.map,用下面几步命令完成
# 第二条没有出错,接着运行,然后造vcf2plink.ok

0.前置命令 生成samples.plink.map
$ vcftools --vcf /share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/vcf_filter/samples.gatk.snp.m2M2.mis0.2.mac3.4dtv.vcf  --plink --out /share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/PTS_analysis/admixture_dir/samples.plink

1.下面的命令用来显示map文件里染色体有哪些
# 如果没有按自然顺序排序,结果会比较奇怪
$ cd analysis/PTS_analysis/admixture_dir
$ a="start" && for i in `awk '{print $2}' samples.plink.map | awk -F ':' '{print $1}'`;do if [ $a != $i ];then echo $i ;a=$i;fi;done

2.对samples.plink.map进行手动替换
# map文件第一列以染色体编号做序号

# 以/share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/PTS_analysis/admixture_dir/samples.plink.map为例
# perl -pi -e "s/0\tGardenia10/10\tGardenia10/g" samples.plink.map # ok
# perl -pi -e 's/0\tGardenia9/9\tGardenia9/g' samples.plink.map # ok
# sed -i  's/0\tGardenia8/8\tGardenia8/g' samples.plink.map # ok

# 使用for循环批量处理
$ for i in {1..11};do echo sed -i  "'s/0\tGardenia$i:/$i\tGardenia$i:/g'" samples.plink.map;done > sed.sh # 显示ok
$ sh sed.sh

3.对samples.plink.map进行自然排序 -n
$ mv samples.plink.map tmp.samples.plink.map
$ cat tmp.samples.plink.map | sort -k 1 -n > samples.plink.map

4.运行第二条命令

5.
$ touch /share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/.flag_dir/vcf2plink.ok

6.继续运行主流程
$ perl /share/nas6/zhouxy/pipline/pop-pip/current/pop_pip_v5.pl -i ref_reseq_config.yaml -o analysis
```

## 5.4 vcffiles2plink.cmds 同5.3
```bash
# vcffiles2plink.cmds出错

1.复制samples.plink.map
$ cp analysis/PTS_analysis/admixture_dir/samples.plink.map analysis/PTS_analysis/pca_dir/samples.plink.map

2.运行第二部分命令

3.
$ touch /share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/.flag_dir/plink2mkbed.ok

4.继续运行主流程
$ perl /share/nas6/zhouxy/pipline/pop-pip/current/pop_pip_v5.pl -i ref_reseq_config.yaml -o analysis
```

## 5.5 fstpi无结果(直接跳过即可)
```bash
# analysis/.cmds_dir/selective_sweep_filter.failed.cmds出错
三组只有一个样品的组间互相对比无结果

$ touch /share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/.flag_dir/selective_sweep_filter.ok

$ perl /share/nas6/zhouxy/pipline/pop-pip/current/pop_pip_v5.pl -i ref_reseq_config.yaml -o analysis
```

## 5.6 selective_sweep_region.failed.cmds 同上
```bash
$ touch /share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/.flag_dir/selective_sweep_region.ok

$ perl /share/nas6/zhouxy/pipline/pop-pip/current/pop_pip_v5.pl -i ref_reseq_config.yaml -o analysis
```

## 5.7 selective_sweep_info.failed.cmds 同上
```bash
$ touch /share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/.flag_dir/selective_sweep_info.ok && perl /share/nas6/zhouxy/pipline/pop-pip/current/pop_pip_v5.pl -i ref_reseq_config.yaml -o analysis
```

## 5.8 selective_sweep_anno.failed.cmds 同上
```bash
$ touch /share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/.flag_dir/selective_sweep_anno.ok && perl /share/nas6/zhouxy/pipline/pop-pip/current/pop_pip_v5.pl -i ref_reseq_config.yaml -o analysis
```

## 5.9 step 14 vcf2plink2_for_gwas_input.cmds 同5.3
```bash
# vcf2plink2_for_gwas_input.cmds出错
# 命令1出错,同5.3,但输入文件变成以下文件
# 命令2也会出错,要改map中第一列

0.命令1
# input: analysis/variation_dir/variants_anno_dir/samples.pop.snp.anno.result.vcf
# out: analysis/GWAS_input/samples.plink.map

0.1删除vcf中非染色体的序列

$ mv analysis/variation_dir/variants_anno_dir/samples.pop.snp.anno.result.vcf analysis/variation_dir/variants_anno_dir/backup_samples.pop.snp.anno.result.vcf # 备份

$ grep "##contig=<ID=" analysis/variation_dir/variants_anno_dir/samples.pop.snp.anno.result.vcf | awk -F "##contig=<ID=" '{print $2}' | awk -F "," '{print $1}' > id_contig.txt # 修改,留下需要删除的id

$ a='' && for i in `cat id_contig.txt`;do b=`echo "$a""\|""$i"`;a=$b;done && echo $b # 生成sed '/所需参数/d' file命令中的参数,前面会多一个 \| ,记得删除

$ sed '/参数/d' analysis/variation_dir/variants_anno_dir/backup_samples.pop.snp.anno.result.vcf > analysis/variation_dir/variants_anno_dir/samples.pop.snp.anno.result.vcf
# 例子 sed '/utg11958_G9\|utg18436_G9\|utg19729_G9\|utg7422_G9\|utg8976_G9\|utg9702_G9\|utg9928_G9\|utg1208_G1\|utg18778_G1\|utg3548_G1\|utg6849_G1\|utg7335_G1\|utg7888_G1\|utg9700_G1\|utg10062_G4\|utg13405_G4\|utg14560_G4\|utg16235_G4\|utg18525_G4\|utg18682_G4\|utg18967_G4\|utg2945_G4\|utg33336_G4\|utg42970_G4\|utg6726_G4\|utg80_G4\|utg1669_G2\|utg18871_G2\|utg6786_G2\|utg9385_G2\|utg9405_G2\|utg14488_G11\|utg34055_G11\|utg10629_G7\|utg10757_G7\|utg43111_G7\|utg6563_G7\|utg7405_G7\|utg7585_G7\|utg9635_G7\|utg1693_G8\|utg26320_G8\|utg28933_G8\|utg9041_G8\|utg10452_G5\|utg12633_G5\|utg13932_G5\|utg16492_G5\|utg23470_G5\|utg38007_G5\|utg4968_G5\|utg7033_G5\|utg7599_G5\|utg8857_G5\|utg1478_G3\|utg3749_G3\|utg6883_G3\|utg8759_G3\|utg11280_G6\|utg13378_G6\|utg20794_G6\|utg11850_G10\|utg14608_G10\|utg3856_G10\|utg8859_G10\|utg8952_G10\|utg9527_G10\|utg9722_G10\|utg12567\|utg18663/d' analysis/variation_dir/variants_anno_dir/backup_samples.pop.snp.anno.result.vcf > analysis/variation_dir/variants_anno_dir/samples.pop.snp.anno.result.vcf

1.重新运行第14步
$ sh /share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/.cmds_dir/vcf2plink2_for_gwas_input.cmds
# 继续运行主流程 perl /share/nas6/zhouxy/pipline/pop-pip/current/pop_pip_v5.pl -i ref_reseq_config.yaml -o analysis

2.命令2出错,对samples.plink.map进行手动替换
# map文件第一列以染色体编号做序号
# 使用for循环批量处理
$ cd /share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/GWAS_input
$ for i in {1..11};do echo sed -i  "'s/0\tGardenia$i:/$i\tGardenia$i:/g'" samples.plink.map;done > sed.sh # 显示ok
$ sh sed.sh

3.对samples.plink.map进行自然排序 -n
$ mv samples.plink.map tmp.samples.plink.map
$ cat tmp.samples.plink.map | sort -k 1 -n > samples.plink.map

4.运行第二条命令

5.
$ touch /share/nas1/yuj/project/GP-20220607-4434_GP-20220614-4481_1_20221124/analysis/.flag_dir/vcf2plink2_for_gwas_input.ok

6.继续运行主流程
$ perl /share/nas6/zhouxy/pipline/pop-pip/current/pop_pip_v5.pl -i ref_reseq_config.yaml -o analysis
```

# 六.GWAS
```shell
# 
$ perl /share/nas6/zhouxy/pipline/gwas_vcf_pip/current/gwas_ped_pip_v2.pl -i ddrad_config.yaml -o analysis

```





