---
title: 04叶绿体流程2-2注释-补充部分
description: 
published: true
date: 2023-03-09T03:59:45.244Z
tags: 
editor: markdown
dateCreated: 2023-03-09T03:59:39.696Z
---


		for i in *.gbk ;do echo $i; cp_gbk2ann.pl -i $i -o $i.ann ;done #抽取注释
		cp_annotation_one_gene_by_ref_gbk2.pl -i1 Thainanus_FULLMT.fsa  -i2 ref_ann/gbk/NC_050664.1.gbk -g COX1 #不看蛋白用叶绿体的找基因或者取序列
		cp_add_gene_sequence.pl  -i ref_ann/fasta/NC_050664.1.fasta -p 74-1039:+  #只取序列不看蛋白用叶绿体的就行
		mt_add_gene_sequence.pl  -i ref_che/fasta/NC_038207.1.fasta -p 13784-13834 -g ?  #要看蛋白的话,用线粒体,需要指定密码子表
		blastn -query tmp.fa  -subject *.fsa -outfmt 6 #blastn


cp_add_gene_sequence.pl 根据基因序列查找新的位置
gene_anno]$ cp_add_gene_sequence.pl  -i *.fsa -p 11371-11772:-     -p表示原来的起止
gene_anno]$ cp_add_gene_sequence.pl  -i *.fsa -p 11371-11772 -c     -c表示负链
gene_anno]$ cp_add_gene_sequence.pl  -i *.fsa -p 11371-11772 -c  -a   -a寻找新的起始终止??????


/share/nas6/xul/program/chloroplast/annotation/annotation_one_gene_by_ref_gbk2.pl # 查找单个基因
gene_anno]$ perl /share/nas6/xul/program/chloroplast/annotation/annotation_one_gene_by_ref_gbk2.pl  -i1 *.fsa -i2 ../../../../ref_ann/gbk/MT671485.1.gbk -g  ndhF 

gene_anno]$ vi tmp.fa  发现差异太多,把参考序列写进文件
gene_anno]$ blastn -query tmp.fa  -subject *.fsa -outfmt 6       但是发现只比对上了前1000
gene_anno]$ cp_add_gene_sequence.pl  -s '参考基因的全部序列'
不输入-i  输入-s,和上面同样作用    好像是看一下这段(参考基因的一部分)对应的蛋白序列,把参考对应的蛋白序列全部表示出来

行 288: [>Thu Nov 04 10:07:24 @compute-0-0 gene_anno]$ 把蛋白序列写入tmp

行 293: [>Thu Nov 04 10:07:29 @compute-0-0 gene_anno]$ tblastn -query tmp.fa  -subject *.fsa -outfmt 6   查找到所有可能匹配的位置,tmp.fa是参考基因序列  这次是蛋白   比不上去

gene_anno]$ nucmer ../../../../ref_ann/fasta/
gene_anno]$ nucmer ../../../../ref_ann/fasta/
gene_anno]$ nucmer ../../../../ref_ann/fasta/MT671485.1.fasta  *.fsa 
gene_anno]$ cp_add_gene_sequence.pl  -i ../../../../ref_ann/fasta/MT671485.1.fasta -p 110000 110800 
gene_anno]$ vi tmp.fa 
gene_anno]$ nucmer ../../../../ref_ann/fasta/
gene_anno]$ nucmer ../../../../ref_ann/fasta/
gene_anno]$ nucmer ../../../../ref_ann/fasta/MT671485.1.fasta  *.fsa 
gene_anno]$ mummerplot out.delta 
gene_anno]$ cp_add_gene_sequence.pl  -i ../../../../ref_ann/fasta/MT671485.1.fasta -p 110000 110800 这段比对不上,从参考取出这段序列,写入tmp
gene_anno]$ vi tmp.fa 
gene_anno]$ cp_bowtie.align.pl  -i tmp.fa  -1 ../../../assembly/
gene_anno]$ cp_bowtie.align.pl  -i tmp.fa  -1 ../../../assembly/Gentianella_pygmaea/
gene_anno]$ cp_bowtie.align.pl  -i tmp.fa  -1 ../../../assembly/Gentianella_pygmaea/pseudo/Gentianella_pygmaea/1_Trimmed_Reads/Gentianella_pygmaea.trimmed_P1.fq  -2 ../../../assembly/Gentianella_pygmaea/pseudo/Gentianella_pygmaea/1_Trimmed_Reads/Gentianella_pygmaea.trimmed_P2.fq  -o bow_ndhF & 
														上面那行把参考的这段去和组装物种去bowtie看能不能比对上
行 358: [>Thu Nov 04 10:12:19 @compute-0-0 gene_anno]$ ll bow_ndhF/
行 364: [>Thu Nov 04 10:12:20 @compute-0-0 gene_anno]$ 

gene_anno]$ ll
行 410: [>Thu Nov 04 10:14:54 @compute-0-0 gene_anno]$ cp_add_gene_sequence.pl  -i *.fsa -p 114012-115553 -c  根据基因序列查找新的位置???????

行 440: [>Thu Nov 04 10:15:58 @compute-0-0 gene_anno]$ 23554734 reads; of these:    上面bowtie比对完了
行 459: [>Thu Nov 04 10:17:24 @compute-0-0 gene_anno]$ 

行 474: [>Thu Nov 04 10:17:33 @compute-0-0 bow_ndhF]$ samtools view -F 4 tmp.sam 查看是否比对到
行 516: [>Thu Nov 04 10:18:00 @compute-0-0 bow_ndhF]$ ^C
行 517: [>Thu Nov 04 10:18:00 @compute-0-0 bow_ndhF]$ 
行 518: [>Thu Nov 04 10:18:00 @compute-0-0 bow_ndhF]$ ll
行 524: [>Thu Nov 04 10:18:01 @compute-0-0 bow_ndhF]$ cat map_pair_hits.*  | paste - - - - |cut -f 2 
行 573: [>Thu Nov 04 10:18:25 @compute-0-0 bow_ndhF]$ cat map_pair_hits.*  | paste - - - - |cut -f 2 | perl -lane 'print ">",++$i,"\n$_"'
行 670: [>Thu Nov 04 10:18:48 @compute-0-0 bow_ndhF]$ cat map_pair_hits.*  | paste - - - - |cut -f 2 | perl -lane 'print ">",++$i,"\n$_"' > map.fa   把比对到的序列写入文件
行 671: [>Thu Nov 04 10:18:54 @compute-0-0 bow_ndhF]$ nucmer ../tmp.fa  map.fa  画图
行 672: [>Thu Nov 04 10:19:01 @compute-0-0 bow_ndhF]$ mummerplot out.delta 
行 687: [>Thu Nov 04 10:19:12 @compute-0-0 bow_ndhF]$ nucmer ../tmp.fa  map.fa  -c 1  画图
行 688: [>Thu Nov 04 10:19:15 @compute-0-0 bow_ndhF]$ mummerplot out.delta 
行 703: [>Thu Nov 04 10:19:41 @compute-0-0 bow_ndhF]$ cp_add_gene_sequence.pl  -i ../*.fsa -p 111857 110708 
行 720: [>Thu Nov 04 10:20:20 @compute-0-0 bow_ndhF]$ cp_add_gene_sequence.pl  -i ../*.fsa -p  110708  111857
行 737: [>Thu Nov 04 10:20:29 @compute-0-0 bow_ndhF]$ grep ^C
行 738: [>Thu Nov 04 10:20:39 @compute-0-0 bow_ndhF]$ ll ../../../../assembly/
行 740: [>Thu Nov 04 10:20:39 @compute-0-0 bow_ndhF]$ ll ../../../../assembly/Gentianella_pygmaea/
行 742: [>Thu Nov 04 10:20:39 @compute-0-0 bow_ndhF]$ ll ../../../../assembly/Gentianella_pygmaea/pseudo/Gentianella_pygmaea/1_Trimmed_Reads/Gentianella_pygmaea.trimmed_P  
行 744: [>Thu Nov 04 10:20:39 @compute-0-0 bow_ndhF]$ cp_get_dump_from_fq_with_jellfish.pl -i  ../../../../assembly/Gentianella_pygmaea/pseudo/Gentianella_pygmaea/1_Trimmed_Reads/Gentianella_pygmaea.trimmed_P*  -m 19 & 
														
	上面程序kmer in genome sequence coverage 
行 967: [>Thu Nov 04 10:25:27 @compute-0-0 bow_ndhF]$ grep ATCATATAACCTAATTGTGACATTGTGGAATACTCTCTTAATATCTTTTTGAGCAAGAACTATTGCTATCAAAAAGATTAAATTC K125.dump 
行 998: [>Thu Nov 04 10:25:38 @compute-0-0 bow_ndhF]$ cat map
行 1000: [>Thu Nov 04 10:25:54 @compute-0-0 bow_ndhF]$ cat map.fa 
行 1097: [>Thu Nov 04 10:25:56 @compute-0-0 bow_ndhF]$ grep ATATACCAATCACAAAAACATCTTTTTTTAGGTACACTTTCTTTTTATGGTATTCCACCTTTAGCATGCTTTTGGTCTAA K125.dump 
行 1098: [>Thu Nov 04 10:26:07 @compute-0-0 bow_ndhF]$ samtools view -F 4 tmp.sam  |less -S 



