---
title: 04叶绿体流程4-个性化分析
description: 
published: true
date: 2023-03-09T03:58:58.082Z
tags: 
editor: markdown
dateCreated: 2023-03-09T03:58:52.879Z
---

# 1.基因分化热图 遗传距离计算
```shell
1.把gb文件放到gbk文件夹中

2.获取同源序列
$ perl /share/nas6/xul/program/chloroplast/phytree/gene_tree/find_total_gene.pl  -o gene -i gbk/ -p 
$ perl /share/nas6/xul/program/chloroplast/phytree/gene_tree/find_total_gene.pl  -i gbk/ -p Gentianella_pygmaea  -o gene   
# -p后面参数名字和gbk文件名里面名字都要一致
$ cd gene/mafft/
$ rm rrn* trn* && nu=`ls ../../gbk/*.gbk |wc |awk '{print $1}'` && rm `grep -c ">" * |grep -v ":$nu"|cut -d ":" -f 1 `

3.计算每个基因的pairwise distance
$ for i in *.fasta ;do echo Rscript /share/nas6/xul/program/chloroplast/pairwise_distance/pairwise_distance.R $i `basename $i .fasta`.distance.xls ;done > pairwise_distance.sh && thread.pl pairwise_distance.sh  63

4.整理结果
$ perl /share/nas6/xul/program/chloroplast/pairwise_distance/get_result.pl *.distance.xls  Gentianella_pygmaea >  all.dis.xls
$ realpath all.dis.xls # 这个文件修改第一行的物种名,不要登陆号

5.作图
$ Rscript /share/nas6/xul/program/chloroplast/pairwise_distance/pheatmap.R all.dis*.xls && convert -density 300 dis.heatmap.pdf  dis.heatmap.png
```

# 2.预测转录后编辑
```shell
perl  /share/nas6/xul/program/mt2/annotation/pipline/src/get_cds_seq_use_gene_annotation_file.pl  -i final_gene_annotaion.info  -f *_FULLCP.fsa -p Gentianella_pygmaea -o tmp_for_prep # 线粒体

perl /share/nas6/xul/program/mt2/annotation/pipline/src/get_cds_seq_use_gene_annotation_file_for_chl.pl -i final_gene_annotaion.info  -f *_FULLCP.fsa -p Gentianella_pygmaea -o tmp_for_prep # 叶绿体

生成文件tmp_for_prep/Gentianella_pygmaea.cds.for_perp-cp.txt

打开http://prep.unl.edu/上传文件

下载文件并解压 tar xf Results.tar.gz 

perl /share/nas6/xul/program/mt2/annotation/pipline/src/get_edit_site_info.pl -i *_Edit_Sites.txt -p Gentianella_pygmaea -o  final_rna_edit_info.xls # 合并表格
```

# 3.SSR
## 3.1流程里ssr及ssr引物设计
```shell
1.SSR分析(/share/nas1/yuj/project/GP-20211130-3777-1_20220325/ssr_analysis下有misa.ini misa.pl)

perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/src/gb2tbl.pl -i 物种.gbk -o tbl
# gbk转tbl

perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/src/tbl_stat.exon.pl -i tbl/物种.tbl -f 物种.fasta -o 物种.exon.intron.stat.xls
# 统计各基因外显子内含子的情况

perl misa.pl 物种.fasta cpSSR
# misa分析ssr,生成cpSSR.misa

perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/etc/misa/src/stat.ssr.pl -i cpSSR.misa -t tbl/物种.tbl -ir 83137-109333,127168-153364 -o1 ssr.info.xls -o2 ssr.stat.xls
# 解析成更容易看的格式 stat是对info的统计

perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/etc/misa/src/plot.ssr.stat.pl -i ssr.stat.xls -o ./ssr_analysis/ 
# 统计后画图
# region.barssr	ssr在不同分区上的内含子外显子间区的分布情况
# region.pie	ssr在不同分区上比例饼状图

perl /share/nas6/xul/program/chloroplast/ssr/draw_ssr.pl -i number.tmp -o ssr.number.bar
# ssr.number.bar	各类ssr计数柱状图


2.引物设计(4286给定ssr 设计其引物)

perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/etc/misa/src/p3_in.pl cpSSR.misa 物种.fasta ./ssr_analysis/
# 生成cpSSR.ssr.p3in	
# 前置文件cpssr.misa(ID 编号 产物长度 开始 终止)

	单纯设计引物，可以直接从填写cpSSR.ssr.p3in开始

perl /share/nas6/xul/program/chloroplast/ssr/ssr_primer_designer.pl -i cpSSR.ssr.p3in -t 20 -o ../ssr_analysis
# 生成cpSSR.ssr.p3out
# 前置文件cpSSR.ssr.p3in  可手动填写,标签详情见 http://www.chenlianfu.com/?p=284 	primer3设计引物详解

perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/etc/misa/src/p3_out.pl cpSSR.ssr.p3out cpSSR.misa cpSSR.ssr.primer.xls 
# 解析成更容易看的格式
```
## 3.2pcr引物设计(当时是3321项目全长基因组来设计)
```shell
比对:
cat ref/fasta/*.fasta > all.fasta
mafft --auto --quiet --thread 30 all.fasta > all.aln #静默比对
	mafft --auto --thread 30 all.fasta > all.aln #有提示信息
/share/nas6/xul/soft/trimal/trimal-1.4.1/source/trimal -in allinone.aln -out all.trim.aln	
把非保守位点替换成N	perl /share/nas6/xul/program/chloroplast/blast/mafft/aln2indel_tmp.pl
进一步把非连续的保守位点替换成N  我的脚本


设计:
	1.填cpssr.misa文件 ID 编号 产物长度 开始 终止,生成引物设计软件的输入文件
	perl primer3_in.pl cpSSR.misa 物种名 .../Personalisation_analysis/pcr

	2.上一步生成cpSSR.ssr.p3in,或者手动填写,标签详情见 http://www.chenlianfu.com/?p=284 	primer3设计引物详解
	perl /share/nas6/xul/program/chloroplast/ssr/ssr_primer_designer.pl -i cpSSR.ssr.p3in -t 20 -o .../Personalisation_analysis/pcr

	3.合并设计好的输出结果
	perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/etc/misa/src/p3_out.pl cpSSR.ssr.p3out cpSSR.misa  cpSSR.ssr.primer.xls 
```
## 3.3查找多态性好的ssr(3672项目答疑)
```shell
	脚本路径
	/share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/etc/MISA/misa.pl
	perl /share/nas6/xul/program/mt2/phytree/gene_tree/src/fasta2line.pl
	参数设置
	def 1-10 2-6 3-5 4-5 5-5 6-5	int 100
	1-10,2-5,3-4,4-3,5-3,6-3
	TBtools功能
	SSRMiner && Batch Target Region Primer Design

	1.简单共线性看一下,应该没必要
	2.比对
	mafft --auto --quiet --thread 30 allinone.fa > all.aln
	3.perl misa.pl 物种.fasta cpssr(输出文件前缀,自定义)
		for i in `ls *.fasta`;do echo $i; perl misa.pl $i ${i%.fasta};done
	4.perl /share/nas6/xul/program/chloroplast/blast/mafft/aln_ssr2marker.pl -i all.aln -d misa_result/ -o ssr20220509.xls
```
## 3.4批量设计ssr引物
```shell
cd misa_result(包含所有.fasta文件)

for i in `ls *.fasta`;do echo $i; perl misa.pl $i ${i%.fasta};done
# 生成misa

for i in *.fasta;do echo $i;mkdir ../${i%.fasta}_ssr_analysis && echo 'done';done
# 生成目录

for i in `ls *.fasta`;do echo $i; perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/etc/misa/src/p3_in.pl ${i%.fasta}.misa $i ../${i%.fasta}_ssr_analysis/ ;done
或者for i in *.fasta;do echo $i;perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/etc/misa/src/p3_in.pl ${i%.fasta}.misa $i ../${i%.fasta}_ssr_analysis/ ;done
# 生成cpSSR.ssr.p3in	
# 前置文件cpssr.misa(ID 编号 产物长度 开始 终止)

(可选)cd ../ && rm */cpSSR.ssr.p3out && cd misa_result
# 删除所有目录里已有的p3out

for i in `ls *.fasta`;do echo $i; perl /share/nas6/xul/program/chloroplast/ssr/ssr_primer_designer.pl -i ../${i%.fasta}_ssr_analysis/cpSSR.ssr.p3in -t 20 -o ../${i%.fasta}_ssr_analysis/ ;done
或者for i in *.fasta;do echo $i;perl /share/nas6/xul/program/chloroplast/ssr/ssr_primer_designer.pl -i ../${i%.fasta}_ssr_analysis/cpSSR.ssr.p3in -t 20 -o ../${i%.fasta}_ssr_analysis/ ;done
# 生成cpSSR.ssr.p3out
# 前置文件cpSSR.ssr.p3in  可手动填写,标签详情见 http://www.chenlianfu.com/?p=284 	primer3设计引物详解

for i in *.fasta;do echo $i;perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/etc/misa/src/p3_out.pl ../${i%.fasta}_ssr_analysis/cpSSR.ssr.p3out ${i%.fasta}.misa ../${i%.fasta}_ssr_analysis/cpSSR.ssr.primer.xls ;done
# 解析成更容易看的格式
```
