---
title: 04叶绿体流程2-1注释
description: 
published: true
date: 2023-03-09T03:58:51.366Z
tags: 
editor: markdown
dateCreated: 2023-03-09T03:58:43.251Z
---

# 程序 路径
```shell
/share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/
/share/nas1/xul/program/chloroplast/cds_analysis/src/
cp_annotation_one_gene_by_ref_gbk2.pl -i1 fasta/MW148820.1.fasta  -i2 ref/gbk/MH042531.1.gbk -g ndhf # 查找单个基因
perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/src/to_genbank_for_juelei.pl # 蕨类生成gbk文件
```
# 1.选择GBK
```shell

1.下载参考
perl /share/nas6/xul/program/chloroplast/bin/cp_get_genbank_form_ncbi_with_ACCESSION.pl -i   -o

2.提取注释
for i in *.gbk ;do echo $i; cp_gbk2ann.pl -i $i -o $i.ann ;done

2ann-
```

参考挑选标准:
cp_Find_fewer_genes.pl  -i xxxx.ann  # 查看关键信息
rps12 3段
ycf1 2个
有时候选同属更顺利
rpl2可能有3段,也就是有内含子,但是有时候两段中间没有碱基,这个时候三段其实就是两段

# 2.注释
## 2.1第一遍注释
```shell

1.配置文件
20210918]$ python3 /share/nas1/yuj/script/chloroplast/get_ann_cfg.py # 在项目编号目录下直接运行
或者cp /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/cp.anno.config.yaml ann.cfg  

# 打开要修改

2.注释
$ perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/bin/chloroplast_annotaion.pl.v2.1.pl -d analysis/assembly/ -i ann.cfg 

# 如果gbk格式不对,需要先用排序的程序重排一遍注释文件(忘了指啥,应该是final_gene_info2文件)
```
## 2.2修改注释
```shell
0.xul版
# 取序列看蛋白
cp_add_gene_sequence.pl  -i *.fsa -p 74-1039:+  
cp_add_gene_sequence.pl  -s ATCGTTGGAACC            
# 查找单个基因
cp_annotation_one_gene_by_ref_gbk2.pl -i1 fasta/MW148820.1.fasta  -i2 ref/gbk/MH042531.1.gbk -g ndhf
blastn -query t  -subject *.fsa -outfmt 6 
tblastn -query t  -subject *.fsa -outfmt 6 

1.根据ann.log先去查看final,取序列看蛋白
cp_add.py  -i *.fsa -p 134809-135306:+
cp_add_gene_sequence.pl  -i *.fsa -p 

2.不对的话再查看tmp查找  查看基因排列
$ grep -i 'psbI' final_gene_annotaion.info2
$ for i in `grep -i 'psbI' final_gene_annotaion.info2 | awk '{print $2}'` ;do cp_add.py  -i *.fsa -p $i;done   > psbI_ann.log
# $ grep 'CDS' ref_ann/gbk/登录号.gbk.ann
# $ for i in `grep 'CDS' ref_ann/gbk/登录号.gbk.ann | awk '{print $2}'` ;do mt_add.py -n 5 -i ref_ann/fasta/登录号.fasta -p $i;done   > ref_ann.log
$ vim cox1
$ tblastn -query cox1 -subject *.fsa -outfmt 6

3.根据记录的笔记检查数量
提取假基因脚本或者gbk搜索pseudo
$ cp_get_some_gene_seq.pl  -i gbk/xxx.gbk -p  > pseudo.fa

4.检查结果
$ cp_change_annotation.v2.pl -i final_gene_annotaion.info2 -o final_gene_annotaion.info && cp_Find_fewer_genes.pl -i final_gene_annotaion.info
# 排序 perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/src/change_annotation.v2.pl -i final_gene_annotaion.info2 -o final_gene_annotaion.info			
# 检查 perl /share/nas6/xul/program/chloroplast/bin/cp_Find_fewer_genes.pl -i final_gene_annotaion.info
```
**注释检查:**
    1.ndhD rpl2  psbL 有时候起始子不是常规起始子,转录时是把中间的c变成u/t,因此在注释结果里加标识符1以便通过流程
    2.trnH-GUG  没有内含子,去掉汇总后表格里的星号,出现该情况是因为起点原因导致此基因跨首尾多了；分号
	3.注意ir区基因是否对称,如3431 里面trna-ile在ira区没了,只在irb有
    4.rps12要注意有俩个外显子,要额外注意选择的参考,也就是三段,数量2个
    5.rps19跨反向重复区很多的话,末尾有对应的假基因,这种情况下共2个,否则1个
    也存在俩rps19基因的情况(如3937 exl龙血树 俩rps19完整位于反向重复区)
    6.rps15应该有1个
    也存在俩rps15(如4086 小黑麦 俩rps15完整位于反向重复区)
    7. ycf15应该是2个或没有
    8.ycf1两个,一真5000一假1000
**PS:**
	1.如果没有注释结果,看看是不是给的gbk有问题,给成了ann文件
	2.correct_ann.log 对ir区的矫正不一定对,通过看注释结果
	3.anno.log2标准含错误输出,看看有没有没找到的基因,一般没有
	4.注意参考cds trna个数  做过的陆生植物一般37 38 39trna  86 87 88cds
## 2.3批量操作
```shell
$ for i in $( ls /share/nas1/yuj/project/GP-20220111-3951_20220215/analysis/annotation/Hibiscus_sabdariffa/gene_anno/gbk) ;do echo $i
$ for i in gbk/*.gbk ;do echo $i; cp_annotation_one_gene_by_ref_gbk2.pl -i1 *.fsa  -i2 /share/nas1/yuj/project/GP-20220111-3951_20220215/analysis/annotation/Hibiscus_sabdariffa/gene_anno/$i -g ndhD;done
$ for i in *.fasta ;do echo $i;cp_ssc.pl -i $i -o $i.fa;done
```

## 2.4第二遍注释
```shell
$ perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/bin/chloroplast_annotaion.pl.v2.1.pl -i ann.cfg -d analysis/assembly/

$ perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/annotation/src/to_genbank_for_juelei.pl # 蕨类生成gbk
```
# 3.整理结果
```shell
perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/cp_pip_dir.pl -i analysis  -o complete_dir # 标准结果

:set nu  vim显示行数
```
# 4.整理报告    
```shell
1.修改报告配置文件
cp /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/html_report/cp.report.cfg report.cfg 

2.标准+高级分析
perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/html_report/report2xml.yelvti.pl -id complete_dir -cfg report.cfg
```
# 5.补充
## 5.1 注释
```shell
perl /share/nas6/xul/program/chloroplast/annotation/Annotation.pl -i1 /share/nas1/yuj/project/GP-20221018-5070_20221031/analysis/annotation/Corynandra_viscosa/gene_anno/Corynandra_viscosa_FULLCP.fsa -i2 /share/nas1/yuj/project/GP-20221018-5070_20221031/analysis/annotation/Corynandra_viscosa/gene_anno/NC_053524.1.gbk -i3 /share/nas1/yuj/project/GP-20221018-5070_20221031/analysis/annotation/Corynandra_viscosa/gene_anno/gene_annotaion.info.bak -o /share/nas1/yuj/project/GP-20221018-5070_20221031/analysis/annotation/Corynandra_viscosa/gene_anno -no  1>/share/nas1/yuj/project/GP-20221018-5070_20221031/analysis/annotation/Corynandra_viscosa/gene_anno/anno.log 2>/share/nas1/yuj/project/GP-20221018-5070_20221031/analysis/annotation/Corynandra_viscosa/gene_anno/anno.log2
# 会生成 _gene_annotaion.info2  _gene_annotaion.info 俩文件
```
## 5.2 圈图
```shell
1.linux分步骤(暂不能用)
$ Rscript /share/nas1/yuj/script/chloroplast/R/Chloroplot.R $outdir/$sample/ogdraw $outdir/$sample/gene_anno/$sample.gbk $sample && convert -density 300 $outdir/$sample/ogdraw/$sample.circular.pdf  $outdir/$sample/ogdraw/$sample.dpi300.circular.png
# 图片转换单拎出来
convert -density 300 *.circular.pdf  $sample.dpi300.circular.png

2.windows分步骤
# 以线粒体为例,cp/mt步骤一样
# Rscript E:\OneDrive\jshy信息部\Script\chloroplast\R\Win_Chloroplot.R $outdir/$sample/ogdraw $outdir/$sample/gene_anno/$sample.gbk $sample cp/mt
$ Rscript E:\OneDrive\jshy信息部\Script\chloroplast\R\Win_Chloroplot.R F:\  F:\4939_202301\Ustilago_esculenta_MT10.gbk Ustilago_esculenta_MT10 mt
$ 复制pdf到服务器
$ convert -density 600 *.circular.pdf  Ustilago_esculenta_MT10.dpi300.circular.png

3.圈图见4086项目txt(暂时不用)
圈图20220506
$ ir xxx.gbk
$ python3 /share/nas1/yuj/script/chloroplast/annotation/ogdraw.py -h
dpi300那张

4.流程里对这部分的文字描述
#-----4.3.2 叶绿体基因组图谱
&EMPTY_TAG('h3','4.3.2 叶绿体基因组图谱','','type1');
&EMPTY_TAG('p','使用Chloroplot（https://irscope.shinyapps.io/Chloroplot/）制作叶绿体基因组图谱，如下图所示：','type1');
if(glob "$indir/04Annotation/*/ogdraw/*dpi300.circular.png"){
	&piclist("图$pid 叶绿体基因组图谱",'注：正向编码的基因位于圈内侧，反向编码的基因位于圈外侧。内部的灰色圈代表GC含量。',"$indir/04Annotation/*/ogdraw/*dpi300.circular.png");    
}else{
	&piclist("图$pid 叶绿体基因组图谱",'注：正向编码的基因位于圈内侧，反向编码的基因位于圈外侧。内部的灰色圈代表GC含量。',"$indir/04Annotation/*/ogdraw/*.circular.png");
}
```
## 5.3 质控（覆盖度）
```shell
# 主程序
$ perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/asmqc/cpDNA_asmqc.pl -i 组装结果  -1 clean*1 -2 clean*2 -r 参考.gbk -p sample -q 物种.gbk  -o asmqc_dir

Bin="/share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/asmqc"

# cmd2 perl $Bin/src/gbk2fa.pl -i $refgbk -o $outdir/refseq.fa
$ perl $Bin/src/gbk2fa.pl -i $refgbk -o refseq.fa
# 生成refsseq.fa 命令3要用

# cmd3 perl $Bin/src/dotplot.pl -r $outdir/refseq.fa -q $infile -o $outdir -p cmp
$ perl $Bin/src/dotplot.pl -r refseq.fa -q $infile -o . -p cmp
# 生成共线性图

# cmd4 perl $Bin/src/annotation_cpDNA_genbank.pl -i $infile -r $refgbk -e sample -o $outdir/cmp.gbk
$ perl $Bin/src/annotation_cpDNA_genbank.pl -i $infile -r $refgbk -e sample -o cmp.gbk
# 生成cmp.gbk 和组装的gbk相比里面名称变成了sample

# cmd5 perl $Bin/src/cmp_genebank.pl -r $refgbk -q $querygbk -o $outdir/cmp.genecov_stat.xls
$ perl $Bin/src/cmp_genebank.pl -r $refgbk -q $querygbk -o cmp.genecov_stat.xls

# cmd8 perl $Bin/src/cp2circos.pl -g $outdir/cmp.gbk -b $outdir/mapping/$prefix.sort.bam -o $outdir/coverage_plot
$ perl $Bin/src/cp2circos.pl -g cmp.gbk -b mapping/*.sort.bam -o coverage_plot
# 如果只生成mapping,运行这个程序,生成覆盖度图片  核心程序

# cmd9 perl $Bin/src/coverage_stat.pl -i $outdir -o $outdir -p $prefix
$ perl $Bin/src/coverage_stat.pl -i . -o . -p sample
# 生成一些统计文件 如果提示缺少文件,就是缺该组装物种的gbk
```

```shell
# cmd8 perl $Bin/src/cp2circos.pl -g $outdir/cmp.gbk -b $outdir/mapping/$prefix.sort.bam -o $outdir/coverage_plot
$ perl $Bin/src/cp2circos.pl -g cmp.gbk -b mapping/*.sort.bam -o coverage_plot
# 会生成覆盖度的svg png

# 修改中间的物种名,把asmqc_dir/coverage_plot/sample.karyotype.txt中的名字改短,注意用下划线连接,不要有空格
circos="/share/nas6/zhouxy/biosoft/circos/current/bin/circos" && $circos -conf coverage_plot/sample.circos.conf
```

# 6.批量操作
```shell
# 批量显示测序数据量
for i in $( ls /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/analysis/assembly/) ;do echo $i; ll /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/analysis/assembly/$i/pseudo/$i/1_Trimmed_Reads/$i.trimmed_P1.fq && ll /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/analysis/assembly/$i/pseudo/$i/1_Trimmed_Reads/$i.trimmed_P2.fq ;done

# 批量显示比对结果
for i in $( ls /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/analysis/assembly/) ;do echo $i; blastn -query /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/ref_ass/all_ref.fa  -subject /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/analysis/assembly/$i/pseudo/$i/1_Trimmed_Reads/uni/assembly.fasta -outfmt 6 ;done

# 测序数据质控
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/annotation/src/fastq_qc.pl -i analysis/samples.reads.txt -o complete_dir/01Rawdata 

# 批量质控 线粒体
for i in $( ls /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/analysis/assembly/) ;do echo $i; perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/asmqc/mtDNA_asmqc.pl -i /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/ref_ass/fasta/HQ857211.1.fasta  -1 /share/nas1/seqloader/xianliti/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/2.clean_data/$i/$i_*_R1_001.fastq.gz -2 /share/nas1/seqloader/xianliti/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/2.clean_data/$i/$i_*_R2_001.fastq.gz -p Gallus_gallus -q /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/ref_ass/gbk/HQ857211.1.gbk  -o /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/analysis/annotation/$i/asmqc -g 2 -r /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/ref_ass/gbk/MT800385.1.gbk;done

# 自动换行查看
perl /share/nas6/pub/pipline/genome-assembly-seq/mitochondrial-genome-seq/v2.0/asmqc/mtDNA_asmqc.pl -i ref_ass/fasta/HQ857211.1.fasta -1 /share/nas1/seqloader/xianliti/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/2.clean_data/$i/$i_*_R1_001.fastq.gz -2 /share/nas1/seqloader/xianliti/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/2.clean_data/$i/$i_*_R2_001.fastq.gz -p Gallus_gallus -q /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/ref_ass/gbk/HQ857211.1.gbk -o /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/analysis/annotation/$i/asmqc -g 2 -r /share/nas1/yuj/project/20211229/GP-20211206-3811-1_longyanxueyuan_9samples_dongwu_xianliti/ref_ass/gbk/MT800385.1.gbk

# 批量质控 叶绿体
for i in $(ls /share/nas1/yuj/project/GP-20210924-3463-1_20220413/analysis/assembly/1/pseudo/1/Final_Assembly/ref_ass/fasta);do echo $i;perl /share/nas6/pub/pipline/genome-assembly-seq/chloroplast-genome-seq/v1.2/asmqc/cpDNA_asmqc.pl -i /share/nas1/yuj/project/GP-20210924-3463-1_20220413/analysis/assembly/1/pseudo/1/Final_Assembly/ref_ass/fasta/$i -1 /share/nas1/seqloader/yelvti/GP-20210924-3463-1_qinghaiminda_zhoulaosh_5samples_yelvti/data/2.clean_data/1/1_S0_L000_R1_000.fastq.gz -2 /share/nas1/seqloader/yelvti/GP-20210924-3463-1_qinghaiminda_zhoulaosh_5samples_yelvti/data/2.clean_data/1/1_S0_L000_R2_000.fastq.gz -p ${i%.1.fasta} -q /share/nas1/yuj/project/GP-20210924-3463-1_20220413/analysis/assembly/1/pseudo/1/Final_Assembly/ref_ass/gbk/${i%.fasta}.gbk  -o /share/nas1/yuj/project/GP-20210924-3463-1_20220413/analysis/assembly/1/pseudo/1/Final_Assembly/${i%.1.fasta}asmqc -r /share/nas1/yuj/project/GP-20210924-3463-1_20220413/analysis/assembly/1/pseudo/1/Final_Assembly/ref_ass/gbk/NC_053552.1.gbk;done

# 批量复制
for i in $(ls /share/nas1/yuj/project/GP-20210924-3463-1_20220413/analysis/assembly/1/pseudo/1/Final_Assembly/ref_ass/fasta);do echo $i;cp /share/nas1/yuj/project/GP-20210924-3463-1_20220413/analysis/assembly/1/pseudo/1/Final_Assembly/${i%.1.fasta}asmqc/coverage_plot/sample.png /share/nas1/yuj/project/GP-20210924-3463-1_20220413/analysis/assembly/1/pseudo/1/Final_Assembly/png/${i%.1.fasta}.png;done
```